ouroboros-consensus/docs/VersioningSchemeDecision.md

```diff
@@ -0,0 +1,286 @@
+
+Cons:
+
+- `main` versions and release versions are not distinguished.
+
+- Cutting a release requires assessing all the changes on `main` since the last release.
+
+## Proposal FallingEdgePatch
+
+A `C`-level PR doesn't alter the `main` version.
+Each `A`-level PR must update the `main` version from `A.B.C` to `(A+1).B.0` unless `A.0.0` is already greater than the previous release.
+Each `B`-level PR must update the `main` version from `A.B.C` to `A.(B+1).0` unless `A.B.0` is already greater than the previous release.
+
+To cut a release from a commit COMMIT on `main` that declares `A.B.C`, announce COMMIT.
+Also immediately update the `main` version to `A.B.(C+1)`.
+
+Cons:
+
+- `main` versions and release versions are not distinguished.
+
+- Some `main` commits will declare version `A.B.(C+1)` even if that version is never officially released or is created only later by backporting _different_ (patch) commits to a previous release branch.
+
+## Proposal Parity
+
+Each `main` version `A.B.C` has an odd `B`.
+Each release version `A.B.C` has an even `B`.
+This is similar to the GHC Team's scheme.
+
+PRs do not alter the `main` version.
+
+To cut a release from a commit COMMIT on `main` that declares version `A.B` (where `B` is necessarily odd), announce a new non-`main` commit that extends COMMIT merely to declare version `X.Y = next(A.B)` (where `Y` is necessarily even).
+Also immediately update the `main` version to `X.(Y+1)`.
+
+Cons:
+
+- Commits on `main` that only include patches since the previous release would still spuriously include the `Y+1` increment in the `B` dimension.
+
+## Proposal NonZero
+
+Each `main` version is degenerate; eg it is always version `0`.
+Each release version is the usual `A.B.C`.
+
+PRs do not alter the `main` version.
+
+To cut a release from a commit COMMIT on `main` (that necessarily declares version `0`), announce a new non-`main` commit that extends COMMIT merely to declare the version to be `next(A.B).0`, or `A.B.(C+1)` depending on what has changed on `main` since the previous release.
+
+Note that the degenerate versions could carry information.
+EG they could be just a single number.
+The only requirement is that they are inherently distinguished from release versions.
+
+Cons:
+
+- `main` versions would always be less than some older released version, which could cause confusion (among people and/or tools).
+
+- Cutting a release requires assessing all the changes on `main` since the last release.
+
+## Proposal Dimension
+
+FYI
+
+```
+Prelude Data.Version> makeVersion [1,2] < makeVersion [1,2,0]
+True
+```
+
+Each `main` version has only two dimensions: `A.B`.
+Each release version has at least three `A.B.C`, where `C` can be `0`.
+
+PRs do not alter the `main` version.
+
+To cut a release from a commit COMMIT on `main` that declares version `A.B`, announce a new non-`main` commit that extends COMMIT merely to declare version `A.B.0`.
+Also immediately update the `main` version to `next(A.B)`.
```

Comment:
This reads like you are releasing version `A.B.0`, _then_ updating the `main` version to `next(A.B)`. If so, how do you decide whether to increment `A` or `B`? There would be no change yet since the release.

I think you probably mean deciding it _before_ the release, and either release `A+1.B.0` or `A.B+1.0`. But then I don't know what's the purpose of the commit that merely declares version `A.B.0`.